src/Uno.UI/UI/Xaml/Controls/Grid/Grid.cs

```diff
@@ -1315,6 +1315,14 @@ protected override XSIZEF MeasureOverride(XSIZEF availableSize)
 		//------------------------------------------------------------------------
 		protected override XSIZEF ArrangeOverride(XSIZEF finalSize)
 		{
+			// UNO NRE FIX
+			if (HasGridFlags(GridFlags.DefinitionsChanged))
+			{
+				// A call to .Measure() is required before arranging children
```

Comment:
Why doesn't the original grid code have this ?

Reply:
Also, what's the circumstance where arrange is called without a previous measure?

Reply:
Those are valid questions. I don't know why original code doesn't have that.  I suspect it's also failing in WinUI, but the exception is going nowhere...

And for the circumstances, I think it's something very specific in the app where we encounter the bug: for some reason, the `.Arrange()` is called before the `.Measure()`. I didn't dig to the faulty code, but I confirmed that's what is happening in this bug.

@jeromelaban @davidjohnoliver

Reply:
Can you also add a unit test to validate this ?